roles/system_ldap/tasks/main.yml

```diff
@@ -1,97 +1,121 @@
     regexp: '^AllowUsers pulsys'
     line: '# AllowUsers pulsys'
   notify: restart sshd
 
-- name: system_ldap | configure pam to enable mkhomedir
-  ansible.builtin.shell: realm permit "{{ ad_user | default('almasftp@pu.win.princeton.edu') }}"
-  become: true
-  tags:
-    - add_sftp_user
-  when:
-    - running_on_server
-  ignore_errors: true
-
-- name: system_ldap | create alma directory drop
-  ansible.builtin.file:
-    path: "{{ item }}"
-    state: directory
-    recurse: true
-    owner: "{{ almasftp_user }}"
-    group: "domain users@pu.win.princeton.edu"
-    mode: 0775
+- name: System_ldap | install necessary packages
+  ansible.builtin.apt:
+    name: "{{ item }}"
+    state: present
+    update_cache: true
   loop:
-    - "/alma/bursar"
-    - "/alma/datasync_processing"
-    - "/alma/fund_adjustment"
-    - "/alma/invoice_status"
-    - "/alma/invoices"
-    - "/alma/people"
-    - "/alma/pod"
-    - "/alma/publishing"
-    - "/alma/recap"
-    - "/alma/scsb_renewals"
-    
-- name: system_ldap | grant {{ deploy_user }} access to alma directory
-  ansible.builtin.user:
-    name: "{{ deploy_user }}"
-    group: "domain users@pu.win.princeton.edu"
-    append: true
-    
-- name: system_ldap | create aspace directory drop
+    - adcli
+    - krb5-user
+    - libnss-sss
+    - libpam-sss
+    - sssd
+    - sssd-tools
+
+- name: System_ldap | configure Kerberos
+  ansible.builtin.blockinfile:
+    path: /etc/krb5.conf
+    mode: "0644"
+    create: true
+    block: |
+      [logging]
+        default = FILE:/var/log/krb5libs.log
+        kdc = FILE:/var/log/krb5kdc.log
+        admin_server = FILE:/var/log/kadmind.log
+
+      [libdefaults]
+        dns_lookup_realm = false
+        dns_lookup_kdc = true
+        ticket_lifetime = 24h
+        renew_lifetime = 7d
+        forwardable = true
+        rdns = false
+        default_realm = {{ ad_domain | upper }}
+        default_ccache_name = KEYRING:persistent:%{uid}
+
+      [realms]
+        {{ ad_domain | upper }} = {
+          kdc = {{ ad_domain_controller }}
+          admin_server = {{ ad_domain_controller }}
+        }
+
+      [domain_realm]
+        .{{ ad_domain | lower }} = {{ ad_domain | upper }}
+        {{ ad_domain | lower }} = {{ ad_domain | upper }}
+
+- name: System_ldap | create sssd.conf
+  ansible.builtin.template:
+    src: sssd.conf.j2
+    dest: /etc/sssd/sssd.conf
+    mode: "0644"
+
+- name: System_ldap | enable and start sssd service
+  ansible.builtin.service:
+    name: sssd
+    state: started
+    enabled: true
+
+- name: System_ldap | join the domain
+  ansible.builtin.command: realm join {{ ad_domain }} -U {{ ad_admin_user }} --install=/
+  environment:
+    KRB5_CONFIG: /etc/krb5.conf
+  changed_when: false
+  when: running_on_server
+
+- name: System_ldap | verify domain join
+  ansible.builtin.command: realm list
+  register: realm_list
+  changed_when: false
+  when: running_on_server
+
+- name: System_ldap | debug realm list output
```

Comment:
This will show the output to the user, but not stop the playbook if that output is wrong. Do we want to verify the domain join? Or was that put in for testing purposes?

Reply:
testing.